src/new/sendTransaction.ts

```diff
@@ -91,123 +118,83 @@ class SendTransaction extends EventEmitter {
+        tokenSet.add(HTR_UID);
         output.token = HTR_UID;
-      }
 
-      if (!(output.token in tokensData)) {
-        tokensData[output.token] = getDefaultData();
-      }
-    }
-
-    // Get tokens array (HTR is not included)
-    const tokens = Object.keys(tokensData).filter((token) => {
-      return token !== HTR_UID;
-    });
-
-    for (const output of this.outputs) {
-      let dataOutput;
-      if (output.type === OutputType.DATA) {
         // Data output will always have value 1 (0.01) HTR
-        dataOutput = {
+        txData.outputs.push({
           type: OutputType.DATA,
-          data: output.data,
+          data: output.data.toString('hex'),
           value: 1,
-          tokenData: 0
-        };
+          authorities: 0,
+          token: output.token,
+        });
       } else {
-        let tokenData;
-        if (output.token === HTR_UID) {
-          // HTR
-          tokenData = 0;
-        } else {
-          tokenData = tokens.indexOf(output.token) + 1;
-        }
-
-        const addressObj = new Address(output.address, { network: this.network });
+        const addressObj = new Address(output.address, { network });
+        tokenSet.add(output.token);
 
-        dataOutput = {
+        txData.outputs.push({
           address: output.address,
           value: output.value,
           timelock: output.timelock ? output.timelock : null,
-          tokenData,
+          authorities: 0,
+          token: output.token,
           type: addressObj.getType(),
-        };
+        });
       }
-
-      tokensData[output.token].outputs.push(dataOutput);
     }
 
-    const walletData = wallet.getWalletData();
-    const historyTxs = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
     for (const input of this.inputs) {
-      const inputTx = historyTxs[input.txId];
-      if (!inputTx || inputTx.outputs.length < input.index + 1) {
+      const inputTx = await this.storage.getTx(input.txId);
+      if (inputTx === null || (!inputTx.outputs[input.index])) {
         const err = new SendTxError(ErrorMessages.INVALID_INPUT);
         err.errorData = { txId: input.txId, index: input.index };
         throw err;
       }
-      const token = inputTx.outputs[input.index].token;
-
-      if (!(token in tokensData)) {
+      const spentOut = inputTx.outputs[input.index];
+      if (!tokenSet.has(spentOut.token)) {
         // The input select is from a token that is not in the outputs
         const err = new SendTxError(ErrorMessages.INVALID_INPUT);
         err.errorData = { txId: input.txId, index: input.index };
         throw err;
       }
-
-      tokensData[token].inputs.push({
-        tx_id: input.txId,
+      txData.inputs.push({
+        txId: input.txId,
         index: input.index,
-        token,
+        value: spentOut.value,
+        token: spentOut.token,
+        address: spentOut.decoded.address!,
+        authorities: transactionUtils.authoritiesFromOutput(spentOut),
```

Comment:
The old sendTransaction facade did not have an authorities or `token_data` because it only allowed common fund outputs (this is ususally done with the delegate operation).
We can change the facade to allow sending authority utxos but this would change the original behavior, so I think it should be on its own PR.

The reason I added the authorities on the inputs is so that the storage method below (`storage.fillTx`) understands what are the inputs of the transaction.